argus/scenarios/service_mock.py

```diff
@@ -88,11 +89,47 @@ def stop_me(self):  # pylint: disable=no-self-use
         cherrypy.engine.exit()
 
 
-class EC2MetadataServiceApp(BaseServiceApp):
-    pass
+class MetadataServiceAppMixin(object):
+    """Common metadata resources."""
+
+    def instance_id(self, *_):
+        return self.scenario._server['id']
+
+    def local_hostname(self, *_):
+        return self.scenario.instance_server()['name'][:15].lower()
+
+    def public_keys(self, *_):
+        return self.scenario.public_key().strip()
 
 
-class CloudstackMetadataServiceApp(BaseServiceApp):
+class EC2MetadataServiceApp(BaseServiceApp, MetadataServiceAppMixin):
```

Comment:
The mixin should be in the left, because if we will use a super call in the mixin, we'll have a bad time.